### PR TITLE
Fix for details page in web UI coming up as blank

### PR DESCRIPTION
**Pull request by Samhita-sopho**
#### Description

For issue #4205, for a flow that connects to mitm.it the details throws an error. Removed that and added an extra case in Details.js

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

**Comment by Samhita-sopho** [comment]
I am not sure where to add tests if any and looked at the failed tests , I am confused on how they are related to the web UI.

File: web/src/js/__tests__/components/FlowView/DetailsSpec.js [hunk 1]
```diff
@@ -46,5 +47,8 @@ describe('Details Component', () => {
         let details = renderer.create(<Details flow={tflow}/>),
             tree = details.toJSON()
         expect(tree).toMatchSnapshot()
+        details = renderer.create(<Details flow={tflowServerAddressNull}/>),
+            tree = details.toJSON()
+        expect(tree).toMatchSnapshot()
```
Comment (mhils) [inline web/src/js/__tests__/components/FlowView/DetailsSpec.js:52]: Let's move this into a seperate test, i.e. `it('should render if the server connection is missing', ...)`. That makes things clearer.

File: web/src/js/__tests__/ducks/_tflowServerAddressNull.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+export default function(){
```
Comment (mhils) [inline web/src/js/__tests__/ducks/_tflowServerAddressNull.js:1]: This is a lot of code for what essentially is TFlow without a server address. I would propose we simply instantiate TFlow in our test function and manually unset the address attribute before we render the component?

File: web/src/js/components/FlowView/Details.jsx [hunk 3]
```diff
@@ -136,8 +139,12 @@ export default function Details({ flow }) {
...
             <ConnectionInfo conn={flow.client_conn}/>
 
-            <h4>Server Connection</h4>
-            <ConnectionInfo conn={flow.server_conn}/>
+            {flow.server_conn.address && [
+                    <div>
+                    <h4>Server Connection</h4>
+                    <ConnectionInfo conn={flow.server_conn}/>
+                    </div>
+            ]}
```
Comment (mhils) [inline web/src/js/components/FlowView/Details.jsx:147]: Nit: We can use [`React.Fragment`](https://reactjs.org/docs/fragments.html) and don't need a new div. :)

**Review by mhils — COMMENTED** [review]
Thanks for the PR!

This looks great overall, but there are a few improvements we can make. Comments below. 😃

**Pull request closed without merging**